piptools/repositories/pypi.py

```diff
@@ -450,17 +450,17 @@ def _setup_logging(self) -> None:
 
         # Sync pip's console handler stream with LogContext.stream
         logger = logging.getLogger()
-        for handler in logger.handlers:
-            if handler.name == "console":  # pragma: no branch
-                assert isinstance(handler, logging.StreamHandler)
-                handler.stream = log.stream
-                break
-        else:  # pragma: no cover
+        logger_names = [x.name for x in logger.handlers]
+        if "console" not in logger_names:  # pragma: no cover
             # There is always a console handler. This warning would be a signal that
             # this block should be removed/revisited, because of pip possibly
             # refactored-out logging config.
             log.warning("Couldn't find a 'console' logging handler")
-
+        else:
+            for handler in logger.handlers:
+                if isinstance(handler, logging.StreamHandler):
+                    handler.stream = log.stream
```

Comment:
while this is expected to suppress the error, this would lead to not assigning `log.stream` to the relevant handler anymore as it's now a different type